async-nats/src/jetstream/context.rs

```diff
@@ -261,7 +261,7 @@ impl Context {
                 CreateStreamErrorKind::EmptyStreamName,
             ));
         }
-        if config.name.contains([' ', '.']) {
+        if !is_valid_name(config.name.as_str()) {
```

Comment:
At this point I wander whether stream name would be better as being it's own type instead of a `String` we check at runtime.

Reply:
Consumer and Streams have the same rules around naming for the same reason: both stream name and subject name are part of the `$JS.API` subject.

I'm not sure if it's worth creating a new type here. If yes, it might be nice to have types also for KV and Object Stores etc and that sounds like a separate PR and work :).